### DISCUSSION TITLE
On_Bench and Injury_Reserved variables for the Player class

### DISCUSSION THREAD

**Discussion by romeil** [cwendt94/espn-api#442] [Ideas] [answered]
It would be nice if I could find out if a particular player is on my team's bench or IR.
Let us take this My Team table for example: 

![ESPN Fantasy Screenshot](https://user-images.githubusercontent.com/102480308/221055200-e239884b-7c1c-45b8-aef7-2725fd4736d2.png)

In this scenario, Giannis Antetokounmpo is not on my team's bench. By virtue of this the `on_bench: booelan` variable would return False. As for a player like DeMar DeRozan, the `on_bench: boolean` variable to return True as he is currently on the team's bench. Last of all, due to the fact that Tre Jones is in the IR slot, the `injury_reserved: boolean` variable returns True. Obviously, for a player who is not within this slot, the variable would return False.

To further explicate how the code would look like:
`giannis.on_bench #returns False`
`demar.on_bench  #returns True`
`tre_jones.injury_reserved #returns True`

I truly believe that these variables would be a great addition to your "Player" class and I hope that you see them as a great idea.
Thank you.

**Comment by romeil** [comment, marked as answer]
Edit: I have just realized that there is a `lineupSlot` variable for the "Player" class, making this unneccesary.

**Reply by cwendt94** [reply]
Awesome, that should give you all the information including if the player is in your IR spot!

**Reply by justinsmits** [reply]
hi all, I have been attempting to use the lineupSlot property, but it says that the property isn't available. I just looked and it seems like the property is on the player class object in the espn-api folder, however I just found that it is not on the player class object in the build\lib folder. Sorry, I normally develop in .net so don't really have much python experience yet :)

**Reply by cwendt94** [reply]
How are you referencing the player class? After initializing the League class you should be able to reference it through the league class or using box score functionality. Check out [Player Information](https://github.com/cwendt94/espn-api/wiki/Football-Intro#player-information) section in the wiki for more details.

**Reply by JasonFreeberg** [reply]
Is there an equivalent property on the NFL Player class? I can't seem to find a way to determine if a player is benched or not

**Reply by cwendt94** [reply]
You can through the box_score functionality and the box_player class has `slot_position`. But looking at the data we can also add this to the base class. I will create a PR adding `lineupSlot` to base player as well so don't always need to get from box_score. Now in version v0.37.0
